activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -286,6 +286,10 @@ def add_to_target(record, skip_callbacks = false, &block)
         replace_on_target(record, index, skip_callbacks, &block)
       end
 
+      def target=(record)
```

Comment:
Why is this method necessary?

Reply:
Two reasons:

1) To cast a singular inverse record to an array.
2) To not [load the association](https://github.com/rails/rails/blob/b9c7305dbe57931a153a540d49ae5d469af61a14/activerecord/lib/active_record/associations/association.rb#L77). When setting the inverse on `has_many` we don't want the collection to be fully loaded because there's probably other records that we don't have in memory yet.

I can alternatively replace this with these two overrides:

```
def target=(records)
  @target = records
end

def inversed_from(record)
  self.target = [record]
  @inversed = !!record
end
alias :inversed_from_queries :inversed_from
```

Reply:
I ended up using a `case` here to use `add_to_target(record)` when the inverse isn't an array. The previous approach seemed to execute SQL in cases where we expected queries not to.

Reply:
[38313][1]

[1]: https://github.com/rails/rails/issues/38313#issuecomment-586697352